features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js

```diff
@@ -97,13 +97,38 @@ class APIClient {
         }
         return APIClient.spec.then((resolved) => {
             return (
-                resolved.spec.paths[resourcePath] &&
-                resolved.spec.paths[resourcePath][resourceMethod] &&
-                resolved.spec.paths[resourcePath][resourceMethod].security[0].OAuth2Security[0]
+                resolved.spec.paths[resourcePath]
+                && resolved.spec.paths[resourcePath][resourceMethod]
+                && resolved.spec.paths[resourcePath][resourceMethod].security[0].OAuth2Security[0]
             );
         });
     }
 
+    /**
+     * Get the scopes allowed for updating the given API field. These scopes are defined in publisher-api REST API
+     * definition, under the API resource definition as scope(s) applicable for all the API fields, i.e.
+     *    API:
+     *      x-scopes:
+     *           - apim:api_create
+     * or as scope(s) separately applicable for particular fields, under API resource. i.e.
+     *      businessInformation:
+     *           x-otherScopes:
+     *                - apim:api_publish
+     * @param field
+     * @returns {*}
+     */
+    static geAPIUpdateAllowedFieldScopes(field) {
```

Comment:
can't find usage of this method here ?

Reply:
This was added to use if required to validate per field scopes on the front-end

Reply:
Removed with 0e39028